### PR TITLE
GPIO motor refactor #1 - Remove currentRPM from state struct

### PR DESCRIPTION
**Pull request by randhid**

### CODE REVIEW

**Review by martha-johnston — APPROVED** [review]
tested on bucket rover and everything looks good. Only comment is to remove the lines where `currentRPM` is commented out

File: components/motor/gpio/motor_encoder.go [hunk 1]
```diff
@@ -170,12 +170,12 @@ type EncodedMotor struct {
 // EncodedMotorState is the core, non-statistical state for the motor.
 // Multiple values should be updated atomically at the same time.
 type EncodedMotorState struct {
-	regulated    bool
-	goalRPM      float64 // <= 0 means worker should do nothing
-	currentRPM   float64
+	regulated bool
+	goalRPM   float64 // <= 0 means worker should do nothing
+	// currentRPM   float64
```
Comment (martha-johnston) [inline components/motor/gpio/motor_encoder.go:175]:
```suggestion
```

File: components/motor/gpio/motor_encoder_test.go [hunk 2]
```diff
@@ -157,9 +157,9 @@ func TestMotorEncoder1(t *testing.T) {
 		ctx := context.Background()
 
 		expectedState := EncodedMotorState{
-			regulated:    false,
-			goalRPM:      0,
-			currentRPM:   0,
+			regulated: false,
+			goalRPM:   0,
+			// currentRPM:   0,
```
Comment (martha-johnston) [inline components/motor/gpio/motor_encoder_test.go:162]:
```suggestion
```

File: components/motor/gpio/motor_encoder_test.go [hunk 3]
```diff
@@ -171,9 +171,9 @@ func TestMotorEncoder1(t *testing.T) {
 		})
 
 		newState := EncodedMotorState{
-			regulated:    true,
-			goalRPM:      100,
-			currentRPM:   10,
+			regulated: true,
+			goalRPM:   100,
+			// currentRPM:   10,
```
Comment (martha-johnston) [inline components/motor/gpio/motor_encoder_test.go:176]:
```suggestion
```

**Comment by randhid** [comment]
> tested on bucket rover and everything looks good. Only comment is to remove the lines where `currentRPM` is commented out

Did you check with the control loop as well?

**Comment by martha-johnston** [comment]
> Did you check with the control loop as well?

yep!

**Pull request merged**